sleepylyze/nrem.py

```diff
@@ -1302,137 +1298,131 @@ def exponential_func(x, a, b, c):
-            if chan not in exclude:
-                spindle_psd_norm[chan] = {}
-
-                # specify data to be fit (only data in norm range)
-                incl_freqs = np.logical_or(((self.spindle_psd_concat[chan].index >= norm_range[0][0]) & (self.spindle_psd_concat[chan].index <= norm_range[0][1])),
-                                            ((self.spindle_psd_concat[chan].index >= norm_range[1][0]) & (self.spindle_psd_concat[chan].index <= norm_range[1][1])))
-                pwr_fit = self.spindle_psd_concat[chan][incl_freqs] 
-
-                # set x and y values (convert y to dB)
-                x_pwr_fit = pwr_fit.index
-                y_pwr_fit = 10 * np.log10(pwr_fit.values)
-
-                # fit exponential -- try second fit line if first throws infinite covariance
+        for chan in self.channels:
+            spindle_psd_norm[chan] = {}
+
+            # specify data to be fit (only data in norm range)
+            incl_freqs = np.logical_or(((self.spindle_psd_concat[chan].index >= norm_range[0][0]) & (self.spindle_psd_concat[chan].index <= norm_range[0][1])),
+                                        ((self.spindle_psd_concat[chan].index >= norm_range[1][0]) & (self.spindle_psd_concat[chan].index <= norm_range[1][1])))
+            pwr_fit = self.spindle_psd_concat[chan][incl_freqs] 
+
+            # set x and y values (convert y to dB)
+            x_pwr_fit = pwr_fit.index
+            y_pwr_fit = 10 * np.log10(pwr_fit.values)
+
+            # fit exponential -- try second fit line if first throws infinite covariance
+            try:
+                popt, pcov = curve_fit(exponential_func, xdata=x_pwr_fit, ydata=y_pwr_fit, p0=(1, 0, 1))
+            except (OptimizeWarning, RuntimeError, TypeError):
                 try:
-                    popt, pcov = curve_fit(exponential_func, xdata=x_pwr_fit, ydata=y_pwr_fit, p0=(1, 0, 1))
-                except (OptimizeWarning, RuntimeError, TypeError):
-                    try:
-                        popt, pcov = curve_fit(exponential_func, xdata=x_pwr_fit, ydata=y_pwr_fit, p0=(1, 1e-6, 1))
-                    except (OptimizeWarning, RuntimeError, TypeError) as e:
-                        popt = np.full(3, np.nan)
-                        chans_norm_failed.append(chan)
-                        print(f'scipy.optimize.curvefit encountered error "{e}" on channel {chan}. Normalization skipped for this channel.')
-                        pass
+                    popt, pcov = curve_fit(exponential_func, xdata=x_pwr_fit, ydata=y_pwr_fit, p0=(1, 1e-6, 1))
+                except (OptimizeWarning, RuntimeError, TypeError) as e:
+                    popt = np.full(3, np.nan)
+                    chans_norm_failed.append(chan)
+                    print(f'scipy.optimize.curvefit encountered error "{e}" on channel {chan}. Normalization skipped for this channel.')
+                    pass
 
-                xx = self.spindle_psd_concat[chan].index
```

Comment:
looks like an indent may have been lost here

Reply:
fixed indent loss in commit 5b04c5c